library/ecp_invasive.h

```diff
@@ -76,6 +76,19 @@ int mbedtls_ecp_gen_privkey_mx(size_t n_bits,
 
 #endif /* MBEDTLS_ECP_MONTGOMERY_ENABLED */
 
+#if defined(MBEDTLS_ECP_DP_SECP521R1_ENABLED)
+
+/** Fast quasi-reduction modulo p521 = 2^521 - 1 (FIPS 186-3 D.2.5)
+ *
+ * \param[in,out]   N_p     The address of the MPI to be converted.
+ *                          Must have 2 * N - 1 limbs, where N is the modulus.
```

Comment:
Should we document that it if limbs are `>= 2 * N -1` they are being cropped and not fail?

~~~~~
if (X_limbs > 2 * P521_WIDTH - 1) {
        X_limbs = 2 * P521_WIDTH - 1;
    }
~~~~~

Reply:
I think we probably should make that check more strict and document this accordingly (see my above comment).

Als o we could update the parameter names to match the implementation (my bad, I forgot to do that when I made the change).